### PR TITLE
rearrange files

### PR DESCRIPTION
**Pull request by dwightguth**
Part of a larger restructuring of K files. Don't review yet.

### CODE REVIEW

File: unittests/runtime-io/io.cpp [hunk 1]
```diff
@@ -69,7 +69,7 @@ extern "C" {
 
 int overwriteTestFile() {
   int fd = ::open("test.txt", O_RDWR | O_CREAT | O_TRUNC, S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH);
-  ::write(fd, "hello world!", 12);
+  BOOST_CHECK_EQUAL(::write(fd, "hello world!", 12), 12);
```
Comment (virgil-serbanuta) [inline unittests/runtime-io/io.cpp:72]: Just curious: this is not related to rearranging files, right?

Comment (dwightguth) [reply unittests/runtime-io/io.cpp:72]: Fixes some warnings that were treated as errors on newer compiler versions.
